### ISSUE TITLE
Issue with | and quantifiers

### ISSUE THREAD

**Issue by Grazfather** [mnemnion/mvzr#4]
```zig
const std = @import("std");
const mvzr = @import("mvzr.zig");

pub fn main() !void {
    const string = "derived dedicated cede 0xdeadDEAD";
    const patterns = [_][]const u8{
        // This doesn't seem to use `\b` correctly
        "[de]{2,}\\b",
        // This doesn't work (no support for non-capture groups
        "\\b(?:0x)?[a-fA-F\\d]{2,}\\b",
        // This shows | works
        "A|a",
        // This doesn't work with two quantifiers
        "0x[a-fA-F0-9]{2,}|[a-fA-F0-9]{2,}",
        // This doesn't work with a quantifier after |
        "0x[a-fA-F0-9]|[a-fA-F0-9]{2,}",
        // This works, but only if the string has `0x`, and doesn't capture greedily
        "\\b0x[a-fA-F0-9]{2,}\\b",
    };

    std.debug.print("String '{s}'\n", .{string});
    for (0..patterns.len) |i| {
        std.debug.print("- Pattern '{s}'\n", .{patterns[i]});
        const regex: mvzr.Regex = mvzr.compile(patterns[i]) orelse continue;
        var it = regex.iterator(string);
        while (it.next()) |m| {
            std.debug.print("  - Match {any}\n", .{m});
        }
    }
}
```

Seems that `{2,}` and `|` both work on their own, but if I use the quantifier again it complains

```
✦ ❯ zig run testregex.zig
String 'derived dedicated cede 0xdeadDEAD'
- Pattern '[de]{2,}\b'
  - Match [0..2]: "de"
  - Match [5..7]: "ed"
  - Match [8..10]: "de"
  - Match [15..17]: "ed"
  - Match [19..22]: "ede"
  - Match [25..27]: "de"
- Pattern '\b(?:0x)?[a-fA-F\d]{2,}\b'
error(mvzr): found a modifier on a modifier
error(mvzr): bad string at 9th character

- Pattern 'A|a'
  - Match [13..14]: "a"
  - Match [27..28]: "a"
  - Match [31..32]: "A"
- Pattern '0x[a-fA-F0-9]{2,}|[a-fA-F0-9]{2,}'
error(mvzr): found a modifier on a modifier
error(mvzr): bad string at 33th character

- Pattern '0x[a-fA-F0-9]|[a-fA-F0-9]{2,}'
error(mvzr): found a modifier on a modifier
error(mvzr): bad string at 29th character

- Pattern '\b0x[a-fA-F0-9]{2,}\b'
  - Match [23..27]: "0xde"
```

**Comment by Grazfather** [comment]
Another: `"[de]{2,}\\b"` matches on `detain` despite the `\\b`

**Comment by mnemnion** [comment]
The non-capture group is just unsupported syntax. Since `mvzr` doesn't do captures, I don't intend to support it unless I figure out a good way to do captures (I do have some ideas there). 

I'll take a look at the others. It would be helpful to report possible bugs in the form of a failing Zig test, if only so that I don't have to guess at the expected behavior.

**Comment by mnemnion** [comment]
Trunk has a fix for the qualifier-on-qualifier bug, with test. I'll take a look at the word-boundary problem later.

**Comment by mnemnion** [comment]
Ok, that should fix the `\b` problem (actually a problem with a specific failure path in star matching, but no matter).

Let me know if you find that it covers the intended cases, and I'll package up a new release.

**Comment by Grazfather** [comment]
> The non-capture group is just unsupported syntax. Since mvzr doesn't do captures, I don't intend to support it unless I figure out a good way to do captures (I do have some ideas there).

Yep! I understand, I only showed it because it shows how I discovered the broken behaviour with my second example.

I tested everything and it looks great! Thanks for fixing this up!

**Comment by mnemnion** [comment]
Fix is up as 0.2.3, thanks for the report!

**Comment by Grazfather** [comment]
Thanks for the library and the fix!
